src/admin/controllers/activities.php

```diff
@@ -94,13 +94,70 @@ public function getActivities()
 		}
 		else
 		{
-			$result_arr['success'] = true;
-			$result_arr['data'] = $result;
+			$result_arr['success']       = true;
+			$result_arr['data']          = $result;
 			$result_arr['data']['total'] = $ActivityStreamModelActivities->getTotal();
 		}
 
 		echo json_encode($result_arr);
 
 		jexit();
 	}
+
+	/**
+	 * Function to delete the activity
+	 *
+	 * @return  void
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+
+	public function delete()
+	{
+		$input  = JFactory::getApplication()->input;
+		$client = $input->get('client', '', 'STRING');
+		$id     = JFactory::getApplication()->input->get('cid', array(), 'array');
+
+		if (!is_array($id) || count($id) < 1)
+		{
+			JLog::add(JText::_($this->text_prefix . '_NO_ITEM_SELECTED'), JLog::WARNING, 'jerror');
+		}
+		else
+		{
+			// Get the model.
+			$model = $this->getModel('activity');
+
+			// Make sure the item ids are integers
+			jimport('joomla.utilities.arrayhelper');
+			ArrayHelper::toInteger($id);
+
+			// Remove the activity.
+			if ($model->delete($id))
+			{
+				$this->setMessage(JText::plural($this->text_prefix . '_N_ITEMS_DELETED', count($id)));
+
+				if (isset($client))
```

Comment:
@Mayaank12 
duplicate code, can be moved out of if else and can be written once

Reply:
Done